stan/math/prim/mat/prob/normal_id_glm_lpdf.hpp

```diff
@@ -92,20 +100,101 @@ return_type_t<T_y, T_x, T_alpha, T_beta, T_scale> normal_id_glm_lpdf(
   const auto &y_val_vec = as_column_vector_or_scalar(y_val);
 
   T_scale_val inv_sigma = 1 / as_array_or_scalar(sigma_val_vec);
+  Matrix<T_partials_return, Dynamic, 1> y_minus_mu_over_sigma_mat(N);
+  auto y_minus_mu_over_sigma = y_minus_mu_over_sigma_mat.array();
+
+  operands_and_partials<T_y, T_x, T_alpha, T_beta, T_scale> ops_partials(
+      y, x, alpha, beta, sigma);
+
+  double y_minus_mu_over_sigma_squared_sum;  // the most efficient way to
+  // calculate this depends on
+  // template parameters
 
-  Array<T_partials_return, Dynamic, 1> y_minus_mu_over_sigma
-      = x_val * beta_val_vec;
+#ifdef STAN_OPENCL
+  const int local_size
+      = opencl_kernels::normal_id_glm.make_functor.get_opts().at("LOCAL_SIZE_");
```

Comment:
Can you fix the kernel_cl class so we don't need to grab the local size out of the make_functor?